client/finality-grandpa/src/authorities.rs

```diff
@@ -220,6 +220,37 @@ where
 	pub(crate) fn current(&self) -> (u64, &[(AuthorityId, u64)]) {
 		(self.set_id, &self.current_authorities[..])
 	}
+
+	/// Revert to a specified block given its `hash` and `number`.
+	/// This removes all the authority set changes that where announced after
+	/// the revert point.
+	/// Revert point is identified by `number` and `hash`.
+	pub(crate) fn revert<F, E>(&mut self, hash: H, number: N, is_descendent_of: &F)
+	where
+		F: Fn(&H, &H) -> Result<bool, E>,
+	{
+		let mut filter = |node_hash: &H, node_num: &N, _: &PendingChange<H, N>| {
+			if number >= *node_num &&
+				(is_descendent_of(node_hash, &hash).unwrap_or_default() || *node_hash == hash)
```

Comment:
This is quite a mouthful of condition that pops up the 3rd time in the codebase. Maybe we can give it a name somewhere next to `is_descendent_of`?